### ISSUE TITLE
High CPU load 

### ISSUE THREAD

**Issue by b-seite** [ctubio/Krypto-trading-bot#9]
Perhaps it´s just a question: I noticed that after on our of running my GDAX Bot gets a quiet high CPU load, even if i stop trading.
When i start K the CPU load of node is around 5%. (latest commit on ubuntu 16.04)  
Then 1 our later it rises up to a maximum of 50-70% and in the same time the trading latency rises up from 1-5ms to 50-100ms. The average load is around 40-50%
perhaps it´s normal behavior but i can't imagine that the bot has so much to do.

thanks for an Answer or help

**Comment by Camille92** [comment]
https://github.com/ctubio/Krypto-trading-bot/issues/3

**Comment by ctubio** [comment]
im sorry imm not able to reproduce this.
also is a bit hard to investigate without knowing your machine specs and without having a memory heap snapshot of the application at the time that the issue was reproducing; can you please consider to upload a heap snapshot? (see https://blog.risingstack.com/finding-a-memory-leak-in-node-js/#toolstechnicsyoucanusetofindamemoryleakinnodejs); and what is your machine?

**Comment by Camille92** [comment]
I just step in to confirm that I get super high CPU load because of Gdax (and it doesn't work) using the previous version of WS. It is all fixed when updated to version 3 but it seems that running 'npm install' puts back the old version.


Here is a screenshot of the CPU credit I use for the gdax instance (and other) when I ran an update yesterday night, it ate al my credit (and I didn't check) until today when I realized it, now it rises again.

![capture d ecran 2017-06-19 a 18 16 28](https://user-images.githubusercontent.com/6861590/27294928-78ff6faa-551b-11e7-857b-34a3ed3c7104.png)
